sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -210,6 +212,20 @@ object QueryParsingErrors {
     new ParseException(s"Intervals FROM $from TO $to are not supported.", ctx)
   }
 
+  def mixedIntervalUnitsError(ctx: IntervalContext): Throwable = {
+    new ParseException(
+      "year-month, day-time and non-ANSI interval units cannot be mixed", ctx)
+  }
+
+  def unsupportedIntervalUnitError(ctx: IntervalContext): Throwable = {
+    val supportedUnits =
+      (YearMonthIntervalType.yearMonthFields.map(YearMonthIntervalType.fieldToString) ++
+        DayTimeIntervalType.dayTimeFields.map(DayTimeIntervalType.fieldToString) ++
+        IntervalUtils.nonAnsiIntervalUnits)
+        .mkString("[", ", ", "]")
+    new ParseException(s"Only $supportedUnits are supported for interval units", ctx)
```

Comment:
nit: it's too verbose to list all the supported fields. How about
```
'ABC' is not a valid interval field: INTERVAL 1 ABC
```

Reply:
I noticed that validation for fields is done in `IntervalUtils.stringToInterval` so I'll just add assertion in `AstBuilder.visitInterval`.